src/rejectdups-rust/src/lib.rs

```diff
@@ -0,0 +1,500 @@
+use byteorder::{ByteOrder, LittleEndian};
+use solana_sdk::{
+    account_info::AccountInfo, entrypoint, entrypoint::ProgramResult, info,
+    program_error::ProgramError, account_info::next_account_info, pubkey::Pubkey,
+    hash::Hasher, 
+    sysvar::{self, Sysvar}, rent::Rent,
+};
+use std::mem;
+
+
+pub fn create_address_with_seed(
+    base: &Pubkey,
+    seed: &str,
+    program_id: &Pubkey,
+) -> Pubkey {
+    let mut hasher = Hasher::default();
+    hasher.hashv(&[base.as_ref(), seed.as_ref(), program_id.as_ref()]);
+    Pubkey::new( hasher.result().as_ref() )
+}
+
+// Declare and export the program's entrypoint
+entrypoint!(process_instruction);
+
+// Program entrypoint's implementation
+fn process_instruction<'a>(
+    program_id: &Pubkey,             // Public key of program account
+    accounts: &'a [AccountInfo<'a>], // data accounts
+    instruction_data: &[u8],         // 1 = vote for A, 2 = vote for B
+) -> ProgramResult {
+
+    info!("Rust program entrypoint");
+
+    // Iterating accounts is safer then indexing
+    let accounts_iter = &mut accounts.iter();
+
+    // Get the account that holds the vote count
+    let account = next_account_info(accounts_iter)?;
+
+    // The account must be owned by the program in order to modify its data
+    if account.owner != program_id {
+        info!("Vote account does not have the correct program id");
+        return Err(ProgramError::IncorrectProgramId);
+    }
+
+    // The data must be large enough to hold two u32 vote counts
+    if account.try_data_len()? < 2 * mem::size_of::<u32>() {
+        info!("Vote account data length too small for u32");
+        return Err(ProgramError::InvalidAccountData);
+    }
+
+    // Get the account that checks for dups
+    let check_account = next_account_info(accounts_iter)?;
+
+    // The check account must be owned by the program in order to modify its data
+    if check_account.owner != program_id {
+        info!("Check account does not have the correct program id");
+        return Err(ProgramError::IncorrectProgramId);
+    }
+
+    // The data must be large enough to hold one u32 int
+    if check_account.try_data_len()? < mem::size_of::<u32>() {
+        info!("Check account data length too small for u32");
+        return Err(ProgramError::InvalidAccountData);
+    }
+
+    // The account must be rent exempt, i.e. live forever
+    let sysvar_account = next_account_info(accounts_iter)?;
+    let rent = <Rent as Sysvar>::from_account_info(sysvar_account)?;
```

Comment:
You could get the rent sysvar like this:

```rust
        let rent = &Rent::from_account_info(next_account_info(account_info_iter)?)?;
```

Reply:
idk tho.... don't you need to do sysvar::rent::check_id(sysvar_account.key)
you can't assume it's the rent account, right - they could pass in a fake rent account i think?

Reply:
Yeah, what I was referring to was getting rid of `<Rent as Sysvar>`.  So maybe should be:

```rust
let sysvar_account = next_account_info(accounts_iter)?;
let rent = &Rent::from_account_info(sysvar_account)?;
if ! sysvar::rent::check_id(sysvar_account.key) {
```

Reply:
ah ok